### PR TITLE
Peerpodconfig removal

### PR DESCRIPTION
**Pull request by bpradipt**
This PR removes peerpodconfig controller and instead moves its functionality of advertising node extended resources for peerpods  to CAA daemonset itself.

### CODE REVIEW

**Review by stevenhorsman — COMMENTED** [review]
This seems like a great change in terms of removing code and simplifying things. Do you see any negatives of have the functionality integrated?
I had question from me about the limit default value and I guess we'll need a follow-up operator PR to reflect the removal as well?
Thanks!

File: src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go [hunk 1]
```diff
@@ -123,6 +123,7 @@ func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 		flags.IntVar(&cfg.networkConfig.VXLANMinID, "vxlan-min-id", vxlan.DefaultVXLANMinID, "Minimum VXLAN ID (VXLAN tunnel mode only")
 		flags.StringVar(&cfg.serverConfig.Initdata, "initdata", "", "Default initdata for all Pods")
 		flags.BoolVar(&cfg.serverConfig.EnableCloudConfigVerify, "cloud-config-verify", false, "Enable cloud config verify - should use it for production")
+		flags.IntVar(&cfg.serverConfig.PeerPodsLimitPerNode, "peerpods-limit-per-node", 1, "peer pods limit per node")
```
[the remark this answers is not in the document]
Comment (bpradipt) [reply src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go:126]: That was my thinking. Users will edit it consciously.  I was in two minds whether to set it to 0 or 1, then thought leaving it to 1 which was the default in peerpodconfig `defaultPeerPodsLimitPerNode` 
Do you recommend setting up a higher default (eg. 10 or 20) for better out-of-box user experience ?

[the remark this answers is not in the document]
Comment (stevenhorsman) [reply src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go:126]: I think 1 is okay, but we'll need to ensure out documentation covers this as something important to review (and I guess some guidance of what might be important factors to consider there)

[the remark this answers is not in the document]
Comment (mkulke) [reply src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go:126]: if we set it to 1 per node, we wouldn't be able to run tests in parallel, if we want to pick an arbitrary number, I would set to it to 10

**Review by mkulke — APPROVED** [review]
nit about the number of peerpods per node, other lgtm

**Comment by bpradipt** [comment]
> This seems like a great change in terms of removing code and simplifying things. Do you see any negatives of have the functionality integrated?
I had question from me about the limit default value and I guess we'll need a follow-up operator PR to reflect the removal as well?

I strongly feel our default deployments should include webhook deployment along with the limit/node. Further hardening the resource management aspects as we move towards 1.0.
Operator PR will be next to remove peerpodconfig. 

@mkulke @stevenhorsman what do you prefer as a good default for the number of peerpods per node?

File: src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevenhorsman) [reply src/cloud-api-adaptor/cmd/cloud-api-adaptor/main.go:126]: Yeah, sounds good.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks

**Comment by bpradipt** [comment]
Rebased and pushed, to trigger the e2e CI again.

**Comment by bpradipt** [comment]
The libvirt e2e test got killed due to the time it took. 
As for the govulncheck, there is a vulnerability report for `gob.Decoder.Decode`. Wonder if we should wait for dependabot alerts and handle it accordingly.

@stevenhorsman @mkulke any thoughts?

**Comment by stevenhorsman** [comment]
> The libvirt e2e test got killed due to the time it took. As for the govulncheck, there is a vulnerability report for `gob.Decoder.Decode`. Wonder if we should wait for dependabot alerts and handle it accordingly.
> 
> @stevenhorsman @mkulke any thoughts?

If we need to bump to 1.22.7 for this fix:
```
Vulnerability #1: GO-2024-3106
    Stack exhaustion in Decoder.Decode in encoding/gob
  More info: https://pkg.go.dev/vuln/GO-2024-3106
  Standard library
    Found in: encoding/gob@go1.21.12
    Fixed in: encoding/gob@go1.22.7
```
I'm not sure if dependabot will do this as it bumps packages rather than the go version? I'm happy for this to merge without that fix as then if dependabot hasn't created anything by Monday I can manually bump and try things out?

**Comment by bpradipt** [comment]
> > The libvirt e2e test got killed due to the time it took. As for the govulncheck, there is a vulnerability report for `gob.Decoder.Decode`. Wonder if we should wait for dependabot alerts and handle it accordingly.
> > @stevenhorsman @mkulke any thoughts?
> 
> If we need to bump to 1.22.7 for this fix:
> 
> ```
> Vulnerability #1: GO-2024-3106
>     Stack exhaustion in Decoder.Decode in encoding/gob
>   More info: https://pkg.go.dev/vuln/GO-2024-3106
>   Standard library
>     Found in: encoding/gob@go1.21.12
>     Fixed in: encoding/gob@go1.22.7
> ```
> 
> I'm not sure if dependabot will do this as it bumps packages rather than the go version? I'm happy for this to merge without that fix as then if dependabot hasn't created anything by Monday I can manually bump and try things out?

I didn't read that the vulnerability is in go standard library. So we will need to bump golang.

I will merge this and open a new PR to bump golang version across the project

**Comment by bpradipt** [comment]
Working on the PR to update golang - https://github.com/confidential-containers/cloud-api-adaptor/pull/2030

**Pull request merged**